internal/cli/test.go

```diff
@@ -285,43 +252,68 @@ Specify the API you want this token for with --audience (API Identifer). Additio
 				cli,
 				tenant,
 				client,
-				"", // specifying a connection is only supported for the test login command
+				"", // Specifying a connection is only supported for the test login command.
 				inputs.Audience,
-				"", // We don't want to force a prompt for the test token command
+				"", // We don't want to force a prompt for the test token command.
 				inputs.Scopes,
-				"",
+				"", // Specifying a custom domain is only supported for the test login command.
 			)
 			if err != nil {
-				return fmt.Errorf("An unexpected error occurred when logging in to client %s: %w", inputs.ClientID, err)
-			}
-			if iostream.IsOutputTerminal() {
-				cli.renderer.GetToken(client, tokenResponse)
-			} else {
-				cli.renderer.Output(tokenResponse.AccessToken)
+				return fmt.Errorf("failed to log into the client %s: %w", inputs.ClientID, err)
 			}
+
+			cli.renderer.TestToken(client, tokenResponse)
+
 			return nil
 		},
 	}
 
 	cmd.SetUsageTemplate(resourceUsageTemplate())
 	cmd.Flags().BoolVar(&cli.force, "force", false, "Skip confirmation.")
 	cmd.Flags().BoolVar(&cli.json, "json", false, "Output in json format.")
-	testClientID.RegisterString(cmd, &inputs.ClientID, "")
 	testAudienceRequired.RegisterString(cmd, &inputs.Audience, "")
 	testScopes.RegisterStringSlice(cmd, &inputs.Scopes, nil)
 
 	return cmd
 }
 
-// cleanupTempApplication will delete the specified application if it is marked
-// as a temporary application. It will log success or failure to the user.
-func cleanupTempApplication(isTemp bool, cli *cli, id string) {
-	if isTemp {
-		if err := cli.api.Client.Delete(id); err != nil {
-			cli.renderer.Errorf("unable to remove the default test application", err.Error())
+func selectClientToUseForTestsAndValidateExistence(cli *cli, cmd *cobra.Command, args []string, inputs *testCmdInputs) (*management.Client, error) {
```

Comment:
Shared logic between both test commands, we select an existing client ID or choose to create one for the test.